File: frontend/src/app/services/admin-services/loan-plan.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { LoanPlan } from './loan-plan';

@Injectable({
  providedIn: 'root',
})
export class LoanPlanService {
  private baseUrl = 'http://localhost:8080/loan-mgmt-system/loan-plans';

  constructor(private http: HttpClient) {}

  getLoanPlanList(): Observable<LoanPlan[]> {
    return this.http.get<LoanPlan[]>(`${this.baseUrl}`);
  }

  addLoanPlan(loanPlan: LoanPlan): Observable<any> {
    return this.http.post(`${this.baseUrl}`, loanPlan);
  }

  getLoanPlanById(id: string): Observable<any> {
    return this.http.get(`${this.baseUrl}/${id}`);
  }

  updateLoanPlan(id: string, loanPlan: LoanPlan): Observable<any> {
    return this.http.put(`${this.baseUrl}/${id}`, loanPlan);
  }

  deleteLoanPlan(id: string): Observable<any> {
    return this.http.delete(`${this.baseUrl}/${id}`);
  }
}

File: backend/src/main/java/com/pijus/loanmgmtsystem/services/LoanPlanService.java
package com.pijus.loanmgmtsystem.services;

import java.util.List;
import java.util.Map;
import java.util.Optional;

import com.pijus.loanmgmtsystem.model.LoanPlan;

public interface LoanPlanService {
	
	public LoanPlan addLoanPlan(LoanPlan plan);

	public List<LoanPlan> getAllLoanPlans();

	public Optional<LoanPlan> getLoanPlanById(String id);

	public LoanPlan updateLoanPlan(String id, LoanPlan plan);

	public Map<String, Boolean> deleteLoanPlan(String id);

}
